File: modules/navigation-module/ios/NavigationPresenterFactory.swift
//
//  NavigationServiceFactory.swift
//  
//
//  Created by Damien L Thompson on 2026-09-02.
//

@MainActor
enum NavigationPresenterFactory {
    static func makePresenter(
        view: NavigationModuleViewProtocol,
        service: NavigationServiceProtocol? = nil,
        builder: NavigationBuilderProtocol? = nil
    ) -> NavigationPresenterProtocol {
        NavigationPresenter(
            view: view,
            service: service ?? NavigationService(),
            builder: builder ?? NavigationBuilder()
        )
    }
}

File: modules/navigation-module/android/src/main/java/expo/modules/navigationmodule/NavigationPresenterFactory.kt
package expo.modules.navigationmodule

import android.content.Context

object NavigationPresenterFactory {
  fun makePresenter(
    view: NavigationModuleViewProtocol,
    context: Context,
    service: NavigationServiceProtocol? = null,
    builder: NavigationBuilderProtocol? = null,
  ): NavigationPresenterProtocol {
    val appContext = context.applicationContext
    return NavigationPresenter(
      view = view,
      context = appContext,
      service = service ?: NavigationService(appContext),
      builder = builder ?: NavigationBuilder(),
    )
  }
}
